### DISCUSSION TITLE
How can i upload source code at Programming Exercise

### DISCUSSION THREAD

**Discussion by vinhpns** [ls1intum/Artemis#7805] [General]
I code one source code and want upload this source code for student who join this course. All student will code continue my source code to finish my requirment. But, i cannot file place upload full source. I code source by Java, spring boot

**Comment by MaisiKoleni** [comment]
What have you tried so far? I assume you have the instructor role and want to create an exercise. Changing the source files (be it for template, solution or tests) is done via version control (git). This should be the connected Bitbucket/GitLab instance, and Artemis should display working links to those repositories. You can then edit the contents as usual. In case you are not familiar with git, have a look around the internet, you will find good learning resources there.

The embedded online code editor – from my point of view – should not be your primary tool here; at least we only used it for small adjustments (such as Comments & Spelling mistakes). All my suggestions here are independent of the programming language.

This is the best advice I can offer based on the limited information provided. I assume you tried to place/drop source files ("file place upload") in the exercise or the online editor in your browser and that you intended to edit the template ("student will code continue my source code").
